jdk/src/share/classes/com/sun/crypto/provider/CipherCore.java

```diff
@@ -1227,33 +1225,58 @@ void updateAAD(byte[] src, int offset, int len) {
         cipher.updateAAD(src, offset, len);
     }
 
+    /*
+     * Get system property with privilage.
+     */
     private static String privilegedGetProperty(final String property) {
-        return AccessController.doPrivileged(new PrivilegedAction<String>() {
-            public String run() {
-                return System.getProperty(property);
-            }
+        return AccessController.doPrivileged(
+            (PrivilegedAction<String>) () -> {
+               return System.getProperty(property);
         });
     }
 
+    /* Read the native crypto properties. The value can be YES/NO or TRUE/FALSE
+     * and case insensitive.
+     *
+     * User can specify disable all native crypto and enable only specific native crypto.
+     * For example -Djdk.nativeCrypto=NO -Djdk.nativeCBC=YES.
+     */
     static {
-        String nativeCryptTrace = privilegedGetProperty("jdk.nativeCryptoTrace");
-        String nativeCryptStr   = privilegedGetProperty("jdk.nativeCrypto");
+        String nativeCryptoTrace = privilegedGetProperty("jdk.nativeCryptoTrace");
+        String nativeCryptoStr   = privilegedGetProperty("jdk.nativeCrypto");
         String nativeCBCStr     = privilegedGetProperty("jdk.nativeCBC");
         String nativeGCMStr     = privilegedGetProperty("jdk.nativeGCM");
 
-        useNativeCrypto = Boolean.parseBoolean(nativeCryptStr) || (nativeCryptStr == null);
+        if (nativeCryptoStr != null &&
+            (nativeCryptoStr.toUpperCase().equals("NO") ||
+             nativeCryptoStr.toUpperCase().equals("FALSE"))) {
+            if (nativeCBCStr == null ||
+                nativeCBCStr.toUpperCase().equals("NO") ||
+                nativeCBCStr.toUpperCase().equals("FALSE")) {
+                useNativeCBC = false;
+            }
 
-        if (!useNativeCrypto) {
-            useNativeCBC = false;
-            useNativeGCM = false;
+            if (nativeGCMStr == null ||
+                nativeGCMStr.toUpperCase().equals("NO") ||
+                nativeGCMStr.toUpperCase().equals("FALSE")) {
+                useNativeGCM = false;
+            }
         } else {
-            useNativeCBC = Boolean.parseBoolean(nativeCBCStr) || (nativeCBCStr == null);
-            useNativeGCM = Boolean.parseBoolean(nativeGCMStr) || (nativeGCMStr == null);
+            if (nativeCBCStr != null &&
+                (nativeCBCStr.toUpperCase().equals("NO") ||
```

Comment:
Note that even without this change you can specify NO to disable, since anything but TRUE equates to false. I'm not sure the code should be written to use negative checks, this means you can use anything else such as OFF and its equivalent to TRUE. I think its better to check for YES and TRUE and anything else equates to false.

Reply:
Thank you @pshipton for your comments. Currently, any value other than TRUE/FALSE or YES/NO are ignored and it takes the default value which is TRUE/YES. I think that makes more sense than specifying any value with the property change the value from default to FALSE.  Please comment whether you still prefer to reverse the check from NO/FALSE to YES/TRUE in the above change.